File: 220818/Java/WeirdStringTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class WeirdStringTest {

  @Test
  void solution() {
    WeirdString test = new WeirdString();

    assertEquals("TrY HeLlO WoRlD", test.solution("try hello world"));
    assertEquals("CoDiNg     DoJaNg I My Me MiNe", test.solution("coding     dojang i my me mine"));
    assertEquals("TrY HeLlO WoRlD", test.solution("TRY hello world"));
    assertEquals("TrYhElLoWoRlD", test.solution("tryhelloworld"));
    assertEquals("      ", test.solution("      "));
    assertEquals("TrYhElLoWoRlD      ", test.solution("tryhelloworld      "));
    assertEquals("     TrYhElLoWoRlD", test.solution("     tryhelloworld"));
    assertEquals("A B C D E F G", test.solution("a b c d e f g"));
  }
}

File: 220818/JavaScript/weird-string.test.js
import solution from './weird-string';

test('solution', () => {
  expect(solution('try hello world')).toBe('TrY HeLlO WoRlD');
});

test('solution', () => {
  expect(solution('coding     dojang i my me mine')).toBe('CoDiNg     DoJaNg I My Me MiNe');
});

test('solution', () => {
  expect(solution('TRY hello world')).toBe('TrY HeLlO WoRlD');
});

test('solution', () => {
  expect(solution('tryhelloworld')).toBe('TrYhElLoWoRlD');
});

test('solution', () => {
  expect(solution('      ')).toBe('      ');
});

test('solution', () => {
  expect(solution('tryhelloworld      ')).toBe('TrYhElLoWoRlD      ');
});

test('solution', () => {
  expect(solution('     tryhelloworld')).toBe('     TrYhElLoWoRlD');
});

test('solution', () => {
  expect(solution('a b c d e f g')).toBe('A B C D E F G');
});
